dashboard/test/controllers/registrations_controller_test.rb

```diff
@@ -370,6 +372,18 @@ class RegistrationsControllerTest < ActionController::TestCase
     refute_includes(mail.body.to_s, 'New to teaching computer science')
   end
 
+  test "create new teacher with MailJet enabled sends welcome email" do
+    teacher_params = @default_params.update(user_type: 'teacher', email_preference_opt_in: 'yes')
+    Geocoder.stubs(:search).returns([OpenStruct.new(country_code: 'CA')])
+    MailJet.stubs(:enabled?).returns(true)
+    MailJet.expects(:create_contact_and_send_welcome_email).once
+    assert_creates(User) do
+      post :create, params: {user: teacher_params}
+    end
+
+    assert_empty ActionMailer::Base.deliveries
+  end
+
   test 'create new teacher with es-MX locale sends localized welcome email' do
```

Comment:
You may need to update the i18n tests for localized emails.
As fas as I know, Dani was experimenting with Mailjet-Crowdin integration to provide localized emails, without exposing a security risk on our end (a.k.a. processing markdown as HTML).

Reply:
We have the mailjet to crowdin sync enabled but I haven't requested translation of the email yet because we don't yet have a way to identify what language the email should be sent in. @mgc1194 do you know of a way we could get that information right now? How accurate is it?

Reply:
Would the language the user has during the signup be the one we want to send the email to?
Once the user has selected a language other than English, it gets stored in the language cookie.

Reply:
We want to add a preferred language field to the sign up process in the future but it did not make the cut for this round. Could be something we consider for early next year

Reply:
The sign up language seems like it would be a good proxy, though, at least for the welcome email. I can add that as a user property on the MailJet side if we want to use it

Reply:
yes, it looks like the welcome email is only supported in English or Spanish-Mexico
https://github.com/code-dot-org/code-dot-org/blob/cf842e965eead6830efcff728848132024729377/dashboard/app/mailers/teacher_mailer.rb#L9-L12

Reply:
Does the update look good? (Assuming the strange Drone errors go away)

Reply:
How do we know if a teacher is Spanish-Mexico? What is the logic it is using right now?

Reply:
It looks like the email might have gotten translated: https://crowdin.com/editor/codeorg-markdown/182198/en-esmx?view=comfortable&filter=basic&value=0. I can try syncing it back to MailJet. If we do that then we would need to send over language to send in to MailJet

Reply:
@castro-jorge FYI on this thread

Reply:
> How do we know if a teacher is Spanish-Mexico? What is the logic it is using right now?

The language of the request. I would assume that's the language the site was in when the user signed up.